### PR TITLE
Improve logging performance (4x) via inlining

### PR DESCRIPTION
**Pull request by Cajunvoodoo**
tl;dr: The inline pragmas total out to ~4x performance improvement in the current benchmark.

Updated the benchmark suite to set up the environment outside of the benchmark code. It heavily pollutes the results for what would--in reality--be a one-time (or infrequent) cost. Additionally, 1000 proved too few writes, so it has been upped to 10000 writes.

Carefully added inline pragmas to many functions, using the benchmark suite and profiling results as a reference. There are a few I am unsure of their impact; it would be nice to have more "real-world" examples to benchmark with and explore more code paths. Others may simply never fire because of how GHC decides to inline (fully saturated functions only). The most impactful functions (found via profiling) were given the most attention.

Notable findings while profiling:
- `brackets` is quite expensive for this benchmark. It took up ~15% of runtime. When inlined, this is hidden (without explicit SCC pragmas), but it is slightly disconcerting how much of the runtime that took up. 
- `scribePermitItem` in `Scribe` is expensive. Unsure what can be done about this. It's a simple enough function (most of the time) that it should be simple to inline, but due to its status as a function in a record, I'm not sure if it *can* inline. It's an indirect function call every time, and it's common enough to be somewhat concerning.
- Compared to other logging libraries, performance seemed *alright*. In terms of raw log output, its ~600x slower than `fast-logger`. Of course, `fast-logger` doesn't have the same feature set and doesn't employ the same "scribe" abstraction. In an old reply on reddit about `katip`'s announcement, it was mentioned that it *might* be possible to use `fast-logger` as a backend for `katip`. I'm fairly sure the changes would be invasive, but the performance improvement could be quite high if that route is taken. *shrug*.

### CODE REVIEW

**Comment by jappeace** [comment]
If this really would be such a large improvement, it'd help me a lot because I'm currently facing an issue where katip messages are being dropped.

I've a "hot loop", which runs about 20 times per second producig ~30 messages per run (all with context).

I'm not sure if katip is the source of the cpu leaking, but the messages are definetly being dropped.
I get about 4 instead of dozens per run.

**Comment by MichaelXavier** [comment]
Fantastic work, @Cajunvoodoo! I have uploaded this to hackage as katip-0.8.8.1. Thank you!

**Comment by Bodigrim** [comment]
I got nerd-snipped to look at the performance of `katip`. Some surface findings:
* `mkThreadIdText` is quite slow, instead of `stripPrefix` on `show` it can use `show` on [`fromThreadId`](https://hackage.haskell.org/package/base-4.20.0.1/docs/GHC-Conc-Sync.html#v:fromThreadId) (as long as `base >= 4.19`).
* Also `T.pack . show` is likely to be slow, going through `toStrict . toLazyText . Data.Text.Lazy.Builder.Int.decimal` should be significantly faster. This affects both `ThreadId` and `ProcessId`.

It would be great if someone gets time to implement this.

**Comment by Bodigrim** [comment]
I implemented the ideas above in https://github.com/Soostone/katip/pull/152
